src/Nest/DSL/UpdateSettingsDescriptor.cs

```diff
@@ -1,116 +1,154 @@
 ﻿using System;
 using System.Collections.Generic;
-using System.Linq;
 using Elasticsearch.Net;
 using Newtonsoft.Json;
 
 namespace Nest
-{
+{ 
+	public class SettingNames
```

Comment:
I don't see any settings in `SettingsNames` that are not Updatable though, maybe I need another coffee :smile: 

Where would we use `Unmodifiable` and `Unremovable` ?